src/FuncSharp/Strings/NonEmptyStringExtensions.cs

```diff
@@ -0,0 +1,199 @@
+﻿using System;
+using System.Globalization;
+
+namespace FuncSharp
+{
+    public static class NonEmptyStringExtensions
+    {
+        public static bool SafeEquals(this NonEmptyString value, string other)
+        {
+            if (value is null)
+                return other is null;
+            return value.Value.SafeEquals(other);
+        }
+
+        public static bool SafeNotEquals(this NonEmptyString value, string other)
+        {
+            if (value is null)
+                return other is not null;
+            return value.Value.SafeNotEquals(other);
+        }
+
+        public static string SafeSubstring(this NonEmptyString s, int length)
+        {
+            return s?.Substring(length);
+        }
+
+        public static string SafeSubstring(this NonEmptyString s, int start, int length)
+        {
+            return s?.Substring(start, length);
+        }
+
+        public static string GetOrElse(this IOption<NonEmptyString> option, string otherwise)
+        {
+            if (option.NonEmpty)
+            {
+                return option.GetOrDefault();
+            }
+            else
+            {
+                return otherwise;
+            }
+        }
+
+        public static string GetOrElse(this IOption<NonEmptyString> option, Func<Unit, string> otherwise)
```

Comment:
btw having an `Unit` as an argument in `otherwise` does not make any sense, it is just a relict from Mr. Siroky

Reply:
you need to start somewhere ;-)

Reply:
I'd actually rather just do a big bang and replace `_ =>` with `() =>` inside the whole solution. Otherwise I'd have to go usage by usage.